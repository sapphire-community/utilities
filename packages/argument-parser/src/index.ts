export * from './lib/errors';
export * from './lib/parsers/Args';
export * from './lib/resolvers/emoji';
export * from './lib/structures/Argument';
export * from './lib/structures/Command';
