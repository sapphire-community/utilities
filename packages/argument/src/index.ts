export * from './lib/errors';
export * from './lib/parsers/Args';
export * from './lib/resolvers/emoji';
export * from './lib/structures';
