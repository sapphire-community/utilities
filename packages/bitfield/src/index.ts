export * from './lib/BitField';
export * from './lib/utilities';
