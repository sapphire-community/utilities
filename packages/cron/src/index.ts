export * from './lib/Cron';
export { Time } from './lib/constants';
