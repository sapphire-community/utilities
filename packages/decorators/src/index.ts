export * from './base-decorators';
export * from './djs-decorators';
export * from './piece-decorators';
export * from './utils';
