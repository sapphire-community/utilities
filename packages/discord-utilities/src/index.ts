export * from './lib/regexes';
export * from './lib/limits';
