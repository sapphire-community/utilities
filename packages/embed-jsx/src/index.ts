import type { MessageEmbed, MessageEmbedOptions } from 'discord.js';

export { EmbedJsx } from './lib/EmbedJsx';

declare global {
	// eslint-disable-next-line @typescript-eslint/no-namespace
	namespace JSX {
		interface IntrinsicElements {
			embed: MessageEmbedOptions;
			title: { url?: string };
			description: Record<string, never>;
			field: { name?: string; inline?: boolean };
			timestamp: Record<string, never>;
			footer: { iconURL?: string };
			image: { url: string };
			thumbnail: { url: string };
			author: { url?: string; iconURL?: string };
		}

		type Element = MessageEmbed;
	}
}
