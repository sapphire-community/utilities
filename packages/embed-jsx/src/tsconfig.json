{
	"extends": "../../ts-config/tsconfig",
	"compilerOptions": {
		"rootDir": "./",
		"outDir": "../dist",
		"tsBuildInfoFile": "../dist/.tsbuildinfo",
		"composite": true,
		"typeRoots": [],
		"jsx": "react",
		"jsxFactory": "EmbedJsx.make"
	},
	"include": ["."]
}
