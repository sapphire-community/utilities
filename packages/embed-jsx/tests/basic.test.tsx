import { MessageEmbed } from 'discord.js';
import { EmbedJsx } from '../src';

test('returns a basic embed', () => {
	const embed = <embed></embed>;
	expect(embed).toStrictEqual(new MessageEmbed());
	expect(embed).toMatchSnapshot();
});

test('returns an embed with options', () => {
	const embed: MessageEmbed = <embed color="RED"></embed>;
	expect(embed).toStrictEqual(new MessageEmbed({ color: 'RED' }));
	expect(embed).toMatchSnapshot();
});

test('return an embed with a title', () => {
	const embed = (
		<embed>
			<title>This is the title</title>
		</embed>
	);
	const expected = new MessageEmbed().setTitle('This is the title');
	expect(embed).toStrictEqual(expected);
	expect(embed).toMatchSnapshot();
});

test('return an embed with a title and url', () => {
	const embed = (
		<embed>
			<title url="https://www.github.com/">GitHub</title>
		</embed>
	);
	const expected = new MessageEmbed().setTitle('GitHub').setURL('https://www.github.com/');
	expect(embed).toStrictEqual(expected);
	expect(embed).toMatchSnapshot();
});

test('return an embed with fields', () => {
	const embed = (
		<embed>
			<field>Hey</field>
			<field name="Title">Hello</field>
			<field name="Inline" inline>
				Hey there
			</field>
		</embed>
	);
	const expected = new MessageEmbed().addFields(
		{ name: '\u200B', value: 'Hey' },
		{ name: 'Title', value: 'Hello' },
		{ name: 'Inline', value: 'Hey there', inline: true }
	);
	expect(embed).toStrictEqual(expected);
	expect(embed).toMatchSnapshot();
});

test('return an embed with a footer', () => {
	const embed = (
		<embed>
			<footer>Hey there</footer>
		</embed>
	);
	const expected = new MessageEmbed().setFooter('Hey there');
	expect(embed).toStrictEqual(expected);
	expect(embed).toMatchSnapshot();
});

test('return an embed with a timestamp', () => {
	const embed = (
		<embed>
			<timestamp>{new Date(10)}</timestamp>
		</embed>
	);
	const expected = new MessageEmbed().setTimestamp(new Date(10));
	expect(embed).toStrictEqual(expected);
	expect(embed).toMatchSnapshot();
});

test('return an embed with a description', () => {
	const embed = (
		<embed>
			<description>Hello!</description>
		</embed>
	);
	const expected = new MessageEmbed().setDescription('Hello!');
	expect(embed).toStrictEqual(expected);
	expect(embed).toMatchSnapshot();
});

test('return an embed with an image', () => {
	const embed = (
		<embed>
			<image url="https://www.google.com/" />
		</embed>
	);
	const expected = new MessageEmbed().setImage('https://www.google.com/');
	expect(embed).toStrictEqual(expected);
	expect(embed).toMatchSnapshot();
});

test('return an embed with a thumbnail', () => {
	const embed = (
		<embed>
			<thumbnail url="https://www.google.com/" />
		</embed>
	);
	const expected = new MessageEmbed().setThumbnail('https://www.google.com/');
	expect(embed).toStrictEqual(expected);
	expect(embed).toMatchSnapshot();
});

test('return an embed with an author', () => {
	const embed = (
		<embed>
			<author>Hi mom!</author>
		</embed>
	);
	const expected = new MessageEmbed().setAuthor('Hi mom!');
	expect(embed).toStrictEqual(expected);
	expect(embed).toMatchSnapshot();
});
