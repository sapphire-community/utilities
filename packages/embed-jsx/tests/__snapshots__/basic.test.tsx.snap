// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return an embed with a description 1`] = `
Object {
  "author": null,
  "color": null,
  "description": "Hello!",
  "fields": Array [],
  "footer": null,
  "image": null,
  "thumbnail": null,
  "timestamp": null,
  "title": null,
  "type": "rich",
  "url": null,
}
`;

exports[`return an embed with a footer 1`] = `
Object {
  "author": null,
  "color": null,
  "description": null,
  "fields": Array [],
  "footer": Object {
    "icon_url": undefined,
    "text": "Hey there",
  },
  "image": null,
  "thumbnail": null,
  "timestamp": null,
  "title": null,
  "type": "rich",
  "url": null,
}
`;

exports[`return an embed with a thumbnail 1`] = `
Object {
  "author": null,
  "color": null,
  "description": null,
  "fields": Array [],
  "footer": null,
  "image": null,
  "thumbnail": Object {
    "url": "https://www.google.com/",
  },
  "timestamp": null,
  "title": null,
  "type": "rich",
  "url": null,
}
`;

exports[`return an embed with a timestamp 1`] = `
Object {
  "author": null,
  "color": null,
  "description": null,
  "fields": Array [],
  "footer": null,
  "image": null,
  "thumbnail": null,
  "timestamp": 1970-01-01T00:00:00.010Z,
  "title": null,
  "type": "rich",
  "url": null,
}
`;

exports[`return an embed with a title 1`] = `
Object {
  "author": null,
  "color": null,
  "description": null,
  "fields": Array [],
  "footer": null,
  "image": null,
  "thumbnail": null,
  "timestamp": null,
  "title": "This is the title",
  "type": "rich",
  "url": null,
}
`;

exports[`return an embed with a title and url 1`] = `
Object {
  "author": null,
  "color": null,
  "description": null,
  "fields": Array [],
  "footer": null,
  "image": null,
  "thumbnail": null,
  "timestamp": null,
  "title": "GitHub",
  "type": "rich",
  "url": "https://www.github.com/",
}
`;

exports[`return an embed with an author 1`] = `
Object {
  "author": Object {
    "icon_url": undefined,
    "name": "Hi mom!",
    "url": undefined,
  },
  "color": null,
  "description": null,
  "fields": Array [],
  "footer": null,
  "image": null,
  "thumbnail": null,
  "timestamp": null,
  "title": null,
  "type": "rich",
  "url": null,
}
`;

exports[`return an embed with an image 1`] = `
Object {
  "author": null,
  "color": null,
  "description": null,
  "fields": Array [],
  "footer": null,
  "image": Object {
    "url": "https://www.google.com/",
  },
  "thumbnail": null,
  "timestamp": null,
  "title": null,
  "type": "rich",
  "url": null,
}
`;

exports[`return an embed with fields 1`] = `
Object {
  "author": null,
  "color": null,
  "description": null,
  "fields": Array [
    Object {
      "inline": false,
      "name": "â€‹",
      "value": "Hey",
    },
    Object {
      "inline": false,
      "name": "Title",
      "value": "Hello",
    },
    Object {
      "inline": true,
      "name": "Inline",
      "value": "Hey there",
    },
  ],
  "footer": null,
  "image": null,
  "thumbnail": null,
  "timestamp": null,
  "title": null,
  "type": "rich",
  "url": null,
}
`;

exports[`returns a basic embed 1`] = `
Object {
  "author": null,
  "color": null,
  "description": null,
  "fields": Array [],
  "footer": null,
  "image": null,
  "thumbnail": null,
  "timestamp": null,
  "title": null,
  "type": "rich",
  "url": null,
}
`;

exports[`returns an embed with options 1`] = `
Object {
  "author": null,
  "color": 15158332,
  "description": null,
  "fields": Array [],
  "footer": null,
  "image": null,
  "thumbnail": null,
  "timestamp": null,
  "title": null,
  "type": "rich",
  "url": null,
}
`;
